src/page/mixin/cartridge-behaviour.js

```diff
@@ -1,41 +1,52 @@
-var isFunction = require('lodash/lang/isFunction');
-var dispatcher = require('focus-core').dispatcher;
-var Empty = require('../../common/empty').component;
-module.exports = {
+import {isFunction, isUndefined} from 'lodash/lang';
+import {dispatcher} from 'focus-core';
+import {component as Empty} from '../../common/empty';
+
+export default {
+
     /**
-     * Register the cartridge.
+     * Updates the cartridge using the cartridgeConfiguration.
      */
-    _registerCartridge: function registerCartridge(){
-      this.cartridgeConfiguration = this.cartridgeConfiguration || this.props.cartridgeConfiguration;
-      if(!isFunction(this.cartridgeConfiguration)){
-        this.cartridgeConfiguration = function cartridgeConfiguration(){
-          return {};
+    _registerCartridge() {
+        this.cartridgeConfiguration = this.cartridgeConfiguration || this.props.cartridgeConfiguration;
+
+        if (!isFunction(this.cartridgeConfiguration)) {
+            this.cartridgeConfiguration = () => ({});
+            console.warn(`
+                Your detail page does not have any cartrige configuration, this is not mandarory but recommended.
+                It should be a component attribute return by a function.
+                function cartridgeConfiguration(){
+                    var cartridgeConfiguration = {
+                    summary: {component: "A React Component", props: {id: this.props.id}},
+                    cartridge: {component: "A React Component"},
+                    actions: {components: "react actions"}
+                    };
+                    return cartridgeConfiguration;
+                }
+            `);
+        }
+
+        let cartridgeConf = this.cartridgeConfiguration();
```

Comment:
I think you can use a `const` instead of a `let` because you don't reaffect them.